client/lxd_server.go

```diff
@@ -55,8 +55,6 @@ func (r *ProtocolLXD) UpdateServer(server api.ServerPut, ETag string) error {
 }
 
 // HasExtension returns true if the server supports a given API extension.
-//
-// Deprecated: Use CheckExtension instead.
```

Comment:
Its still deprecated, and places where its used need to be updated.

Reply:
I see a need for both `HasExtension` and `CheckExtension` as the former just tells you if you have it or not and the later gives you either an error or nil.

All the places I've seen `HasExtension` being used was because a nicer/more in-context error message needed to be conveyed to the user.

Reply:
I'd suggest we leave it deprecated to encourage a gradual switch over to the preferred CheckExtension.